File: java/freeman/command/undo/TVOffCommand.java
public class TVOffCommand implements Command{
    TV tv;
    public TVOffCommand(TV tv){
        this.tv=tv;
    }
    public void execute(){
        tv.off();        
    }
    public void undo(){
        tv.on();
        tv.setChannel(7);
        tv.setVolume(10);
    }
}

File: php/oop-study/lesson3/Undo/TVOffCommand.php
<?php
	class TVOffCommand implements Command{
		private $tv;
		public function __construct(TV $tv){
			$this->tv=$tv;
		}
		public function execute(){
			$this->tv->off();//Выключили			
		}
		public function undo(){
			$this->tv->on();//Включили
			$this->tv->setVolume(10);//задали громкость звука
			$this->tv->setChannel(1);//Задали канал
		}
	}
?>